mongo/options/createcollectionoptions.go

```diff
@@ -136,6 +136,11 @@ type CreateCollectionOptions struct {
 	//
 	// This option is only valid for MongoDB versions >= 6.0
 	EncryptedFields interface{}
+
+	// ClusteredIndex is used to create a collection with a clustered index.
+	//
+	// This option is only valid for MongoDB versions >= 5.3
+	ClusteredIndex interface{}
```

Comment:
@qingyang-hu I believe this commit will overlap/create merge conflicts with GODRIVER-2383. If this PR gets merged before your eventual PR for 2383, you may only have to sync specification tests.

Reply:
I think [GODRIVER-2383](https://jira.mongodb.org/browse/GODRIVER-2383) also requires changes in IndexOptions. BTW, why isn't it a *bool here?